spec/models/zip_endpoint_spec.rb

```diff
@@ -99,6 +99,34 @@
       expect(described_class.pluck(:endpoint_name).sort).to eq expected_ep_names
     end
 
+    it 'alerts and continues if a settings entry is not addable (allows e.g. partial initial config of credentials via env var)' do
+      zip_endpoints_setting = Config::Options.new(
+        forthcoming_s3_north_endpoint:
+          Config::Options.new(
+            secret_access_key: 'sdXSDr+asdfe/lkljoWEDCljdE+aTWrefc'
+          )
+      )
+      logger = instance_double(Logger, warn: nil)
+      allow(described_class).to receive(:logger).and_return(logger)
+      allow(Settings).to receive(:zip_endpoints).and_return(zip_endpoints_setting)
+      allow(Honeybadger).to receive(:notify)
+      allow(logger).to receive(:warn)
```

Comment:
Since you have `warn: nil` declared on the double above, I believe you should be able to spy on `warn` without this line.

Reply:
ah good catch, thanks!

Reply:
removed, still seemed to work fine locally:
```diff
% git diff origin/zip_endpoint_seed_error_handling HEAD
diff --git a/spec/models/zip_endpoint_spec.rb b/spec/models/zip_endpoint_spec.rb
index 15ad9a2a..afe458b0 100644
--- a/spec/models/zip_endpoint_spec.rb
+++ b/spec/models/zip_endpoint_spec.rb
@@ -110,7 +110,6 @@ RSpec.describe ZipEndpoint do
       allow(described_class).to receive(:logger).and_return(logger)
       allow(Settings).to receive(:zip_endpoints).and_return(zip_endpoints_setting)
       allow(Honeybadger).to receive(:notify)
-      allow(logger).to receive(:warn)
 
       # run it a second time
       described_class.seed_from_config
```

will merge once CI is green again.